recipe/build-pkg.sh

```diff
@@ -11,10 +11,17 @@ fi
 # Build vanilla version (no avx2), see build-lib.sh
 cmake -G Ninja \
     ${CMAKE_ARGS} \
-    -Dfaiss_ROOT=_libfaiss_generic_stage/ \
-    -DFAISS_ENABLE_GPU=${FAISS_ENABLE_GPU} \
-    -DCMAKE_BUILD_TYPE=Release \
-    -DPython_EXECUTABLE="${PYTHON}" \
+    -Dfaiss_ROOT:PATH=_libfaiss_generic_stage/ \
+    -DFAISS_ENABLE_GPU:BOOL=${FAISS_ENABLE_GPU} \
+    -DCMAKE_BUILD_TYPE:STRING=Release \
+    -DPython_EXECUTABLE:PATH="${PREFIX}/bin/python" \
+    -DPython_ROOT_DIR:PATH="${PREFIX}" \
+    -DPython_INCLUDE_DIR:PATH=$(${PYTHON} -c 'from sysconfig import get_paths; print(get_paths()["include"])') \
+    -DPython_INCLUDE_DIRS:PATH=$(${PYTHON} -c 'from sysconfig import get_paths; print(get_paths()["include"])') \
+    -DPython_LIBRARY:PATH="${PREFIX}/lib/libpython${PY_VER}.dylib" \
+    -DPython_LIBRARIES:PATH="${PREFIX}/lib/libpython${PY_VER}.dylib" \
+    -DPython_NumPy_INCLUDE_DIR:PATH=$(${PYTHON} -c 'import numpy; print(numpy.get_include())') \
+    -DPython_NumPy_INCLUDE_DIRS:PATH=$(${PYTHON} -c 'import numpy; print(numpy.get_include())') \
```

Comment:
This points to the `numpy` in the `build` environment, which is [consistent with `opencv` where cross-compilation is also used]( https://github.com/conda-forge/opencv-feedstock/blob/891639f983996e8f7cae4595b53d7293cd31058b/recipe/build.sh#L125 )